__tests__/costumes.test.js

```diff
@@ -0,0 +1,455 @@
+    it("should create a new row in the table", async () => {
+        //TODO: Fix this open handle
+        console.log("connected");
+        await createTables();
+        const {rows} = await pool.query(`
+            SELECT COUNT(*) FROM costumes;
+        `)
+        const rowsBefore = rows[0].count;
+        await createCostume(
+            "ballroom gown",
+            "adult",
+            "female",
+            "L",
+            "dress",
+            1,
+            150.99
+        );
+        const {rows: rowsAfterAddingCostume} = await pool.query(`
+            SELECT COUNT(*) FROM costumes;
+        `)
+        const rowsAfter = rowsAfterAddingCostume[0].count;
+        expect(rowsBefore).toBe('0')
+        expect(rowsAfter).toBe('1');
+    })
+
+    it("should create a new entry with correct values", async () => {
+        console.log("connected");
+        await createTables();
+        await createCostume(
+            "buttless chaps",
+            "adult",
+            "unisex",
+            "M",
+            "pants",
+            3,
+            75.99
+        );
+        const {rows: [chaps]} = await pool.query(`
+            SELECT * FROM costumes WHERE name='buttless chaps';
+        `);
+        expect(chaps.name).toBe("buttless chaps");
+        expect(chaps.category).toBe("adult");
+        expect(chaps.gender).toBe("unisex");
+        expect(chaps.size).toBe("M");
+        expect(chaps.type).toBe("pants");
+        expect(chaps.stock_count).toBe(3);
+        expect(chaps.price).toBe(75.99);
+    })
+
+    it("should create multiple entries when called multiple times", async () => {
+        console.log("connected");
+        await createTables();
+        await createCostume(
+            "bonnet",
+            "child",
+            "female",
+            "S",
+            "hat",
+            8,
+            14.99
+        );
+        const {rows: [bonnet]} = await pool.query(`
+            SELECT * FROM costumes WHERE name='bonnet';
+        `);
+        await createCostume(
+            "epaulet",
+            "adult",
+            "unisex",
+            "M",
+            "accessory",
+            4,
+            24.99
+        );
+        const {rows: [epaulet]} = await pool.query(`
+            SELECT * FROM costumes WHERE name='epaulet';
+        `);
+        expect(bonnet.name).toBe("bonnet");
+        expect(bonnet.category).toBe("child");
+        expect(bonnet.gender).toBe("female");
+        expect(bonnet.size).toBe("S");
+        expect(bonnet.type).toBe("hat");
+        expect(bonnet.stock_count).toBe(8);
+        expect(bonnet.price).toBe(14.99);
+
+        expect(epaulet.name).toBe("epaulet");
+        expect(epaulet.category).toBe("adult");
+        expect(epaulet.gender).toBe("unisex");
+        expect(epaulet.size).toBe("M");
+        expect(epaulet.type).toBe("accessory");
+        expect(epaulet.stock_count).toBe(4);
+        expect(epaulet.price).toBe(24.99);
```

Comment:
again if you use args objects, you could write a helper function that takes an args object and a query result and confirms the two match. Then you can reuse that helper function every time you are checking.

Reply:
I'll try that out!